packages/modular-scripts/src/typecheck.ts

```diff
@@ -0,0 +1,65 @@
+import isCI from 'is-ci';
+import path from 'path';
+import ts from 'typescript';
+import chalk from 'chalk';
+import getPackageMetadata from './buildPackage/getPackageMetadata';
+import { reportTSDiagnostics } from './buildPackage/reportTSDiagnostics';
+import * as logger from './utils/logger';
+
+export async function typecheck(silent: boolean): Promise<void> {
+  const { typescriptConfig } = await getPackageMetadata();
```

Comment:
Same with this if we're getting the `typescriptConfig` let's hoist it to be re-usable as we'll probably want to plug it through for `start` / `build` in the future as well?